### PR TITLE
AdminUI - Add SearchKit-based screen for Administer Financial Accounts

### PR DESCRIPTION
**Pull request by davejenx**
Overview
----------------------------------------
Add a SearchKit implementation of the "Administer Financial Accounts" screen to the CiviCRM Admin UI extension.

Before
----------------------------------------
Existing core implementation of the "Administer Financial Accounts" screen:

financial-accounts-before

After
----------------------------------------
New SearchKit implementation of the "Administer Financial Accounts" screen:

financial-accounts-after2

Technical Details
----------------------------------------
This required changing the URL used for add, update & delete actions on financial accounts to civicrm/admin/financial/financialAccount/edit, because this SearchKit implementation overrides the path civicrm/admin/financial/financialAccount .

### CODE REVIEW

File: ext/civicrm_admin_ui/managed/SavedSearch_Administer_Financial_Accounts.mgd.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            [
+              'size' => 'btn-xs',
+              'links' => [
+                [
+                  'entity' => 'FinancialAccount',
+                  'action' => 'update',
+                  'join' => '',
+                  'target' => 'crm-popup',
+                  'icon' => 'fa-pencil',
+                  'text' => 'Edit',
```
Comment (Damilare1) [inline ext/civicrm_admin_ui/managed/SavedSearch_Administer_Financial_Accounts.mgd.php:145]: Just like we did to the labels, please add internalisation to the text parameter for the buttons.

Comment (davejenx) [reply ext/civicrm_admin_ui/managed/SavedSearch_Administer_Financial_Accounts.mgd.php]: Agreed and done.

**Comment by Damilare1** [comment]
* General standards
    * Explain ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain))
        * __PASS__ : The goal/problem/solution have been adequately explained in the PR.
    * User impact ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user))
        * __PASS__: The change would be intuitive or unnoticeable for a majority of users who work with this feature.
    * Documentation ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc))
        * __PASS__: The changes do not require documentation.
    * Run it ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))
        * __PASS__: Ran it on the test sites, installed the civicrm_admin_ui and formbuilder extensions, navigated to the path `/civicrm/admin/financial/financialAccount` and tested all the CRUD operations which all worked well.
* Developer standards
    * Technical impact ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech))
        * __PASS__: The change preserves compatibility with existing callers/code/downstream.
    * Code quality ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code))
        * __PASS__: The functionality, purpose, and style of the code seems clear+sensible.
    * Maintainability ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint))
        * __PASS__: The change sufficiently improves test coverage, or the change is trivial enough that it does not require tests.
    * Test results ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test))
        * __PASS__: The test results are all-clear

**Review by Damilare1 — APPROVED** [review]
* General standards
    * Explain ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain))
        * __PASS__ : The goal/problem/solution have been adequately explained in the PR.
    * User impact ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user))
        * __PASS__: The change would be intuitive or unnoticeable for a majority of users who work with this feature.
    * Documentation ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc))
        * __PASS__: The changes do not require documentation.
    * Run it ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))
        * __PASS__: Ran it on the test sites, installed the civicrm_admin_ui and formbuilder extensions, navigated to the path `/civicrm/admin/financial/financialAccount` and tested all the CRUD operations which all worked well.
* Developer standards
    * Technical impact ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech))
        * __PASS__: The change preserves compatibility with existing callers/code/downstream.
    * Code quality ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code))
        * __PASS__: The functionality, purpose, and style of the code seems clear+sensible.
    * Maintainability ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint))
        * __PASS__: The change sufficiently improves test coverage, or the change is trivial enough that it does not require tests.
    * Test results ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test))
        * __PASS__: The test results are all-clear

**Comment by colemanw** [comment]
This looks good. Just a couple small changes I'd recommend and then I think it's done.
1. Set a conditional style on the rows like some of the other PRs to show a disabled style when `is_active = false`.
2. Translating partial sentences broken up by links makes for hard-to-translate strings. Maybe we can re-word the help text a little to make it flow better without mid-sentence links?
3. In the xml paths, the delete link is missing `&reset=1`. I see it's also missing from the old form, but I think that was an oversight.
4. Very minor point - the aff.json file is missing a newline at the end.

**Comment by wmortada** [comment]
> 4\. Very minor point - the aff.json file is missing a newline at the end.

I noticed that with mine as well. This is how CiviCRM generates the file so I kept it as is. Should we alter CiviCRM so that it adds a newline at the end of these files?

**Comment by colemanw** [comment]
Good idea @wmortada - see #24756

**Comment by colemanw** [comment]
@davejenx are you still able to work on this PR?

**Comment by davejenx** [comment]
Hi Coleman, I'd like to, I have a bit of a backlog of urgent stuff to get through after the sprint so it may be a couple of weeks.

**Comment by colemanw** [comment]
Ok I've pushed up a few updates & I think it's good now. The strings are still a bit iffy but we can fix those up anytime; doesn't have to hold up this PR.

**Pull request merged**